Rebase to master

OBattler opened the pull request:
Hello,

I am sorry for this pull request all of a sudden but I saw no other way of contacting you.

I am the 86Box maintainer and I have just discovered your PS/55 implementation today. What is the current state of this implementation? Would it be possible to pull request this to upstream?

Thanks in advance for your time and understanding!

Summary
=======
_Briefly describe what you are submitting._

Checklist
=========
* [ ] Closes #xxx
* [ ] I have discussed this with core contributors already
* [ ] This pull request requires changes to the ROM set
  * [ ] I have opened a roms pull request - https://github.com/86Box/roms/pull/changeme/

References
==========
_Provide links to datasheets or other documentation that helped you implement this pull request._

jriwanek committed:
Initial Micro Solutions CompatiCard support

jriwanek committed:
Corrections to the SixPakPlus

jriwanek committed:
Add Generic MM58167 RTC

jriwanek committed:
General changes from the obattler_202406 branch

Co-Authored-By: Miran Grča <oubattler@gmail.com>

akmed772 commented:
I'm sorry. I have been busy with my other works and missed your comment.

The current code almost works, but there are some issues.

* The Ctrl+Alt+Del key code has been changed because the IBM 5576-002 keyboard assigns the different function for the L-Alt position.
* It does not display characters correctly in OS/2 J1.33.

However, I don't have enough time to fix them now.

I'm using the ROM files 38F6933 and 94X1320 found here.
https://ardent-tool.com/PS55/5550/386.html
https://ardent-tool.com/PS55/video/

I don't know how to upload ROMs and merge to the original master. I'll leave it to you.

Thank you.

jriwanek committed:
Gameport backend work

jriwanek committed:
Config

OBattler commented:
> I'm sorry. I have been busy with my other works and missed your comment.
> 
> The current code almost works, but there are some issues.
> 
>     * The Ctrl+Alt+Del key code has been changed because the IBM 5576-002 keyboard assigns the different function for the L-Alt position.
> 
>     * It does not display characters correctly in OS/2 J1.33.
> 
> 
> However, I don't have enough time to fix them now.
> 
> I'm using the ROM files 38F6933 and 94X1320 found here. https://ardent-tool.com/PS55/5550/386.html https://ardent-tool.com/PS55/video/
> 
> I don't know how to upload ROMs and merge to the original master. I'll leave it to you.
> 
> Thank you.

No problem and thanks for your response! I'll see what I can do.

jriwanek committed:
malloc to calloc

jriwanek committed:
Clean up .available & .poll

jriwanek committed:
Initial code for joystick port instancing

Dizzy611 committed:
Add BIOS selection to Packard Bell PB450.

jriwanek committed:
Cleaning

Dizzy611 committed:
Formatting fixes.

jriwanek committed:
Corrections

jriwanek committed:
Missed fix

Dizzy611 committed:
Remove outdated TODOs.

jriwanek committed:
Merge pull request #5125 from Dizzy611/pb450-bios-selection

Allow selection between PCI and PNP BIOSes on the PB450.

jriwanek committed:
Revert two fixes as batty has that covered

OBattler committed:
Merge pull request #5124 from jriwanek-forks/warnfixes

Random warning fixes and cleaning

unreal9010 committed:
Update machine.h

unreal9010 committed:
Update m_at_slot1.c

unreal9010 committed:
Update machine_table.c

jriwanek committed:
Various improvements & Cleanups

Some ported from ANightly's work

OBattler committed:
Merge pull request #5130 from jriwanek-forks/anightly

Various improvements & Cleanups

OBattler committed:
Merge pull request #5131 from jriwanek-forks/compaticard

Initial Micro Solutions CompatiCard support

OBattler committed:
Merge pull request #5129 from unreal9010/master

Add ABIT BX6

unreal9010 committed:
Fixed a BX6-related typo

OBattler committed:
Merge pull request #5133 from unreal9010/master

Fixed a BX6-related typo

starfrost013 committed:
Implement the rivatimer and cyclical logging from my main NV3 branch so that we can get it reviewed and merged.

starfrost013 committed:
forgot to port over some parts

starfrost013 committed:
Don't punch me in the face, and also fix the compilation, and also fix a stupid bug in log replay.

starfrost013 committed:
Fix incorrect include

starfrost013 committed:
fix the screwed up callbacks

TC1995 committed:
NCR 5380-based changes of the day (January 12, 2025)

1. Sanity check for the SCSI temp_buffer if it's allocated or not.
2. Data reads and writes in non-DMA mode should be accessible only when DMA mode is Idle (as in, no DMA at all, whereas DMA mode will go to the SCSI controllers' callbacks).

OBattler committed:
Merge pull request #5136 from 86Box/tc1995

NCR 5380-based changes of the day (January 12, 2025)

starfrost013 committed:
fix compile by fixing capitalisation

starfrost013 committed:
hopefully the final issue...fix missing prototype

starfrost013 committed:
explicit void...

TC1995 committed:
Okay, maybe for the 53c400 only so.

Based on my tests, the former 5380 commit will apply to the 53c400 only, for now...

OBattler committed:
Merge pull request #5137 from 86Box/tc1995

Okay, maybe for the 53c400 only so.

OBattler committed:
PIC: Signal to the CPU to end the block on IRR read, fixes stalls in waiting for IRQ, fixes #5108.

OBattler committed:
Fix NEC Vx0 INS*/OUS* timings calculation.

OBattler committed:
Fixed the fix.

OBattler committed:
FIX timings of SYSCALL/SYSRET/SYSENTER/SYSEXIT/FXSAVE/FXSTOR.

eddmanx committed:
Update README.md - added the Avalonia86 manager

jriwanek committed:
Merge pull request #5138 from eddmanx/patch-1

Update README.md - added the Avalonia86 manager

OBattler committed:
Reverted the PIC fix as it was apparently a red herring.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

TC1995 committed:
More NCR53c400 fixes (January 13th, 2025)

Getting on my nerves, NCR 5380...

OBattler committed:
Merge pull request #5140 from 86Box/tc1995

More NCR53c400 fixes (January 13th, 2025)

TC1995 committed:
Remove excess logs.

jriwanek committed:
Merge pull request #5141 from 86Box/tc1995

Remove excess logs.

unreal9010 committed:
Added a couple of Quantum Fireball SE/EX drives

OBattler committed:
AT KBC: Fast track command AE (enable keyboard) because the LG MultiNet sends command A7 immediately after it, fixes keyboard lock-ups in its CMOS Setup.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
Merge pull request #5142 from unreal9010/master

Added a couple of Quantum Fireball SE/EX drives

starfrost013 committed:
Move cyclical logging to the new logging system per obat feedback.

OBattler committed:
Acer V35N: Do not add the NVR - it is already added by the Super I/O chip.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
Merge pull request #5135 from starfrost013/phase0

RIVA 128 preparation: Implement high-frequency timer asynchronous to guest CPU and a new logging function to detect repeated patterns of lines

OBattler committed:
SCAMP: Fixed EMS memory states, fixes #1966.

jriwanek committed:
Cleanups in isartc

OBattler committed:
NEAT: Fix EMS and implement Shadow RAM and top 128K of conventional memory on/off, closes #1375.

unreal9010 committed:
Few corrections to Quantum Fireball series

unreal9010 committed:
Update cdrom.h for TEAC CD-532E/NEC CDR-1900A

unreal9010 committed:
Add TEAC CD-532E (32X)/NEC CDR-1900A (32X)

unreal9010 committed:
Update scsi_cdrom.c

unreal9010 committed:
Update cdrom.h

unreal9010 committed:
Fireball EX/CR now complete

unreal9010 committed:
Updated INQUIRY

unreal9010 committed:
Update cdrom.h

OBattler committed:
Merge pull request #5143 from unreal9010/master

Added TEAC CD-532E (32X)/NEC CDR-1900A (32X) CD-ROM drives, completed the EX/CR line of Fireballs and made a few corrections

jriwanek committed:
Correct EV-165A's name

jriwanek committed:
Fix EV-165A config

OBattler committed:
Merge pull request #5144 from jriwanek-forks/ev-165

Quick fixes to the Everex EV-165A

OBattler committed:
Fix maximum amount of RAM on a few NEAT machine where it was set unrealistically high.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

MaxwellS04 committed:
Added a plenty of IDE CD-ROM models

-Added other four ATA-3 models
-Corrections to Western Digital pre-ATA-4 HDD models

MaxwellS04 committed:
Small name fixes to CDR-1900A

MaxwellS04 committed:
Correct the revision names

OBattler committed:
NEAT: More fixes, closes #5145.

MaxwellS04 committed:
Name corrections to LG CRN-8245B

MaxwellS04 committed:
Added other two models (1x DVD/1x CD)

Also updated the revision on CRN-8245B

MaxwellS04 committed:
Added other five models I have forgotten

Also changed the HL-DT-ST CD-ROM to reflect the model revision.

MaxwellS04 committed:
Merge remote-tracking branch 'upstream/master' into CD_ROM_Batch

MaxwellS04 committed:
Added one recently-added ATA-1 HDD model

greblosdier committed:
Update 386_dynarec.c

Changed exec386_dynarec_dyn from inline to noinline to resolve NEW_DYNAREC build issue on Linux x86_64.

OBattler committed:
Merge pull request #5151 from greblosdier/patch-1

Change exec386_dynarec_dyn to noinline in 386_dynarec.c

OBattler committed:
NEAT: Actually properly initialize the shadow RAM mappings, fixes #5148.

OBattler committed:
Merge pull request #5146 from MaxwellS04/CD_ROM_Batch

Added a plenty of IDE CD-ROM models (+other HDD changes)

OBattler committed:
Dynamic recompiler: Only disable inline if not on Windows or MacOS.

OBattler committed:
NEAT: RA0-RA2 defaults and RA0 alt. reset.

jriwanek committed:
CONFIG_BIOS for ibmat (5170)

OBattler committed:
Improved the Linux new dynamic recompiler freeze workaround per patch by Dax.

jriwanek committed:
GlaBIOS for 5150/5160

MaxwellS04 committed:
Added 9 more ATA-4/ATA-5 HDD models

-Small corrections to ATA-4 Western Digital HDD models
-Line fixes to cdrom.h

jriwanek committed:
Initialized ret to 0

OBattler committed:
Merge pull request #5152 from jriwanek-forks/ibmpc

IBM and RTC Changes

MaxwellS04 committed:
Added note on CFS1275A

OBattler committed:
Merge pull request #5153 from MaxwellS04/ATA_Batch

Add 9 more ATA-4/ATA-5 HDD models I have forgotten

MaxwellS04 committed:
A small fix to SV2046D

jriwanek committed:
Merge pull request #5155 from MaxwellS04/ATA_Batch

A small fix to SV2046D

telanus committed:
GlaBIOS for Juko ST

telanus committed:
GlaBIOS for Juko ST

TC1995 committed:
1st set of changes (video-related) of the day (January 16th, 2025)

ATI Mach8-based (add-on only):
Do not override the clone VGA banked mapping with the standard VGA when needed. Fixes issues with banked mapping with ATI 8514/A enabled on cards with clone mappings such as Cirrus, Video7, Paradise/WD, etc.

OBattler committed:
Merge pull request #5157 from 86Box/tc1995

1st set of changes (video-related) of the day (January 16th, 2025)

greblosdier committed:
Fix incorrect chipset for MVI486

Changes the chipset of the Mylex MVI486 from Opti 82C895 to 82C495

greblosdier committed:
Fixes incorrect chipset model for MVI486

Fixes MVI486 to use Opti 495 chipset

OBattler committed:
SCAMP: Reworked a lot of stuff, backfill EMS should now also work.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
The forgotten mem.c/h changes.

OBattler committed:
NEAT: A few clean-ups.

OBattler committed:
Merge pull request #5158 from greblosdier/MVI486-fix-opti495

Fix incorrect Opti chipset for MVI486

jriwanek committed:
Merge pull request #5156 from telanus/master

GlaBIOS for Juko ST

OBattler committed:
NEAT: More fixes, NEAT machines now POST again.

richardg867 committed:
CS423x: Change Control Indirect Address reserved bit readout for VS440FX BIOS

richardg867 committed:
CS423x: Clean up RAM download state machine

richardg867 committed:
CS423x: Add logging, which somehow never existed

richardg867 committed:
CS423x: Refactor EEPROM-less mode

OBattler committed:
FPU: Properly implement INT 10h FPU exception, fixes #5162.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

richardg867 committed:
ISAPnP: Create logical devices that don't exist instead of erroring out

richardg867 committed:
CS423x: Fixes to PnP in EEPROM-less mode

richardg867 committed:
CS423x: Fix CS4236B game port I/O inaccuracy

richardg867 committed:
CS423x: Plumbing for more chips and clean-ups

richardg867 committed:
AD1848: Plumbing for more CS423x chips

TC1995 committed:
The delayed video changes of the night (January 20th, 2025)

Generic SVGA layer:
Added function pointers of the banked mapping for use with add-on cards with their own mapping when the VGA card banked mapping is not active or viceversa (e.g.: XGA).

XGA-1/2:
1. Reimplemented Area Fill and Boundary Mode as best as possible.
2. Fixed conflicts with banked mapping with VGA clones.
3. Fixed inverted colors (again) on accelerated 16bpp mode under OS/2.

Video7 with ATI 8514/A add-on.
Added a workaround (BIOS issue? I don't know) that disables 8514/A mode and reenables VGA mode when needed. Fixes screen freezes and polling issues with various drivers for Windows and others.

telanus committed:
GlaBIOS for Vendex HeadStart Turbo 888-XT

telanus committed:
GlaBIOS for Vendex HeadStart Turbo 888-XT

OBattler committed:
Merge pull request #5169 from 86Box/tc1995

The delayed video changes of the night (January 20th, 2025)

OBattler committed:
Merge pull request #5170 from telanus/master

GlaBIOS for Vendex HeadStart Turbo 888-XT

richardg867 committed:
AD1848: Fix additional data format unlocking which has been wrong this whole time

richardg867 committed:
CS423x: Fix broken codec on CS4236B

richardg867 committed:
AD1848: Add one more CS4235 register access backdoor

TC1995 committed:
Big SCSI bus update of the day, NCR 5380 too (January 20th, 2025)

1. Separate the SCSI bus functions from NCR 5380 into true general purpose SCSI bus functions, allowing use of future legacy scsi controllers.
2. Corrected NCR 5380 chip period for the SCSI controllers based on that chip so that CD-ROM speed is correct enough per speed tests and no more breakage (I hope, report if they are still there, please!) on desyncs.
3. A NCR 5380 software reset involves asserting an IRQ.

OBattler committed:
Merge pull request #5172 from 86Box/tc1995

Big SCSI bus update of the day, NCR 5380 too (January 20th, 2025)

OBattler committed:
NVR: Get/set time functions now take a void pointer, so nvr.h can be included without requiring time.h.

sentogahara committed:
Add the AT&T Globalyst 330 (Pentium/Socket 4)

jriwanek committed:
Fixes to tmacm

jriwanek committed:
Actually close cartridge on error

TC1995 committed:
Video changes of the day (January 26th, 2025)

XGA:
1. Added a proper poll so to have its timings independent of the SVGA core and changes its renderer accordingly, mainly a blank render one.
2. Workaround the Ctrl-Alt-Del reset of Windows 2.x so that it switches back to VGA text mode correctly when using the 0xA000 64K aperture (in vid_svga.c).

SVGA core:
Re-organized the way the different timers are organized so that the XGA and 8514/A pollers can coexist with the SVGA timer without conflicts.

ATI Mach32:
Reworked the true color renderer and made it a Mach32 exclusive due to its differences from the standard one. Fixes wrong rendered colors in NT using the 32bpp renderer in RGB mode.

OBattler committed:
Merge pull request #5180 from 86Box/tc1995

Video changes of the day (January 26th, 2025)

TC1995 committed:
More changes to the 5380 chips (January 26th, 2025)

Apparently the Trantor T130B SCSI controllers has a different way of calculating the timings and removed the scsi_bus_read() calls from the Current SCSI bus status port (Read Port+4).
Fixes NT using said controller as well as CD swapping while maintaining the correct accurate CD speed.

OBattler committed:
Merge pull request #5181 from 86Box/tc1995

More changes to the 5380 chips (January 26th, 2025)

TC1995 committed:
8514/A and S3 minor change (January 26th, 2025)

1. Cosmetic changes.
2. Revert the position of the starting coordinates of the Short Stroke command, fixes some glitches in NT and elsewhere.

OBattler committed:
Merge pull request #5182 from 86Box/tc1995

8514/A and S3 minor change (January 26th, 2025)

jriwanek committed:
Assorted warning fixes

OBattler committed:
Merge pull request #4786 from chungy/f8f12_everywhere

Use F8+F12 to release mouse on all platforms

OBattler committed:
Merge pull request #5177 from rushieda/globalyst330_p5

Add the AT&T Globalyst 330 (Pentium/Socket 4)

OBattler committed:
Assorted fixes, including warning message box for unavailable devices and translation fixes.

OBattler committed:
Russian translation fixes by lemondrops.

OBattler committed:
Radisys EPC-2102: Move it to among the 430HX machines, where it should be.

sentogahara committed:
Add the PC Partner VIA809DS (Socket 7 D.V./VIA VP3)

OBattler committed:
Merge pull request #5184 from rushieda/via809ds

Add the PC Partner VIA809DS (Socket 7 D.V./VIA VP3)

OBattler committed:
The great CD-ROM clean-up and rewrite, fixes #5134.

OBattler committed:
Fixed CD-ROM image mounting crashes and, hopefully, Linux compiles.

jriwanek committed:
Remove incorrect addition to cmakelist.

OBattler committed:
More *nix compile fixes.

OBattler committed:
And another.

OBattler committed:
And yet more *nix compile fixes.

OBattler committed:
Make handle a void pointer on *nix.

OBattler committed:
CD-ROM Image: close log before closing the file, fixes segmentation fault on image unload.

TC1995 committed:
Video changes of the night (January 29th, 2025)

Matrox MGA:
A break from work can save the day all so to say. Fixes bitblt'ing on WinNT4.0/Win2000 using 32bpp modes and possibly other stuff using said Matrox chip.

OBattler committed:
Merge pull request #5188 from 86Box/tc1995

Video changes of the night (January 29th, 2025)

OBattler committed:
Unix SDL: Fix compile-breaking errors.

OBattler committed:
And a missing include file.

OBattler committed:
CD-ROM: No longer fatal when encountering a mode 1 sector on a XA disc - apparently such discs actually exist.

OBattler committed:
It's image_open(), no longer cdrom_image_open().

OBattler committed:
CD-ROM: Correctly return from read sector with the return code from the back-end in case of error, fixes System Shock 2 CD detection.

OBattler committed:
Fixed some CD-ROM commands not stopping audio play when they should have (fixes data playing as audio, producing nasty artifacts, in some games), and increased the media history for all drive types from 4 to 10, also refactored cdrom_read_sector() and moved the last module-wide variables into the cdrom_t struct.

MaxwellS04 committed:
Make the HDD models' speed accurate

Also make some, but few, of HDD model names more accurate.

OBattler committed:
Merge pull request #5189 from MaxwellS04/Faster_HDD

Make HDD models' speeds accurate

OBattler committed:
Merge pull request #5192 from jriwanek-forks/resourceleak

Some fixes to tmacm and carts

jriwanek committed:
Macro Cleaning

OBattler committed:
Merge pull request #5193 from jriwanek-forks/gameports

Initial work for Gameport instancing

OBattler committed:
Merge pull request #5194 from jriwanek-forks/available

Clean up .available & .poll

OBattler committed:
New recompiler: read the two bytes of unaligned 16-bit immediates into the correct IR bytes, fixes Windows 95 Setup GPF'ing.

TC1995 committed:
Minor ATI 8514/A Ultra add-on change of the night (February 1st, 2025)

Actually make it work on NT by setting bit 0 of the read only config2 reg.

OBattler committed:
Merge pull request #5195 from jriwanek-forks/calloc

malloc to calloc

OBattler committed:
Merge pull request #5197 from jriwanek-forks/warnfixes

Assorted warning fixes

jriwanek committed:
Rename define+device for lotech EMS board

jriwanek committed:
Rename define+device for Realtek 8019AS PNP

jriwanek committed:
Trivial EMS cleanups and a few comments

jriwanek committed:
Fix for building on msys2 clang found by ANightly

OBattler committed:
Merge pull request #5198 from jriwanek-forks/cleanups

Cleanups and a bugfix

OBattler committed:
XGA: Move a variable initialization to IF blocks, fixes segementation fault when pat_src is 8 (eg. in Windows 95 on the MCA XGA).

OBattler committed:
Update CMakeLists.txt

OBattler committed:
Update qt_main.cpp

OBattler committed:
Update qt_main.cpp

OBattler committed:
Update qt_main.cpp

OBattler committed:
Merge pull request #5200 from Cacodemon345/windows-dark-mode

Windows dark mode

OBattler committed:
Update qt_main.cpp

OBattler committed:
Update CMakeLists.txt

OBattler committed:
Fix a warning.

OBattler committed:
Merge pull request #5201 from 86Box/darkmode

Windows dark mode

OBattler committed:
CD-ROM: IOCTL now detects medium changes without having to read the host drive's TOC on every operation, improves performance.

jriwanek committed:
More cleanups to device structs

sentogahara committed:
Fix various errors with the machine table for 586-based machines (Part 1)

OBattler committed:
Merge pull request #5206 from rushieda/mtfix586

Fix various errors with the machine table for 586-based machines (Part 1)

richardg867 committed:
Merge branch 'master' of ssh://github.com/86Box/86Box

OBattler committed:
No longer list Socket 8 CPU's for Slot 1 machines, closes #5196.

richardg867 committed:
Merge branch 'master' of ssh://github.com/86Box/86Box

richardg867 committed:
CS423x: Finish implementing the missing chip types

richardg867 committed:
CS423x: Add on-board CS4236 to the VS440FX

richardg867 committed:
CS423x: Add proper on-board CS4236B to the Intel AP440FX

Default PnP ID patches have been removed as the AP440FX BIOS confirms the CSC4236 ID never changed (as later documented on CS4235/9)

OBattler committed:
Merge pull request #5208 from jriwanek-forks/cleanup

Next batch of cleanups

OBattler committed:
Fix the AC flag on 8-bit and 16-bit ADC and SBB instructions.

jriwanek committed:
A couple of cleanups in hdc_ide.c

jriwanek committed:
Add ISA, VLB & PCI second channel IDE devs

richardg867 committed:
CS423x: Add some more missing CS4235 features and locks

richardg867 committed:
CS423x: Add one more missing CS4235 hardware config byte

richardg867 committed:
CS423x: Playback powerdown check fixes

jriwanek committed:
FIFO tagged functions

Co-Authored-By: Miran Grča <oubattler@gmail.com>

richardg867 committed:
CS423x: Add remaining CS4235 extended->control register access paths

jriwanek committed:
Serial IRQ function

Co-Authored-By: Miran Grča <oubattler@gmail.com>

jriwanek committed:
ACCESS.bus

Co-Authored-By: Miran Grča <oubattler@gmail.com>

OBattler committed:
Merge pull request #5212 from jriwanek-forks/idesec

Support for standalone secondary IDE

jriwanek committed:
Seperate Socket 3 and Socket 3 PCI

Co-Authored-By: Miran Grča <oubattler@gmail.com>

jriwanek committed:
Fix warnings in ide_sec

Co-Authored-By: Miran Grča <oubattler@gmail.com>

OBattler committed:
Merge pull request #5213 from jriwanek-forks/idesec

Fix warnings in ide_sec

OBattler committed:
Merge pull request #5214 from jriwanek-forks/43prep

General changes from the obattler_202406 branch

jriwanek committed:
More Socket 3 PCI

jriwanek committed:
Merge pull request #5217 from jriwanek-forks/s3pci

More Socket 3 PCI

richardg867 committed:
Jenkins: Revise MacPorts OpenAL hack for 1.24.2 and hopefully later updates, fixes #5218

jriwanek committed:
Remove duplicate .close in scsi_t228_device

OBattler committed:
Default host_serial_path to empty string instead of NULL, fixes segmentation fault on initilization of serial passthrough with non-initialized string.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
Fixes to the device and machine configuration string getters and accordingly reverted the serial passthrough fix as well as it's no longer needed.

OBattler committed:
Some const changes in the FDD code.

OBattler committed:
The device_t available/poll union is now gone, mouse poll is now set using mouse_set_poll(), and mouse_curr is now also gone.

OBattler committed:
The NVR name is now obtained from the selected machine BIOS type if applicable.

OBattler committed:
Generic Xt memory expansion: Allow starting at 640k.

OBattler committed:
Removed the last remnants of device_t.poll().

OBattler committed:
Added the test mode entry point requested by gloriouscow.

OBattler committed:
Documented three more functions.

sentogahara committed:
Block AMD and Cyrix CPUs from Intel Socket 5/7 motherboards with stock BIOS

OBattler committed:
Merge pull request #5224 from rushieda/mtfix586

Block AMD and Cyrix CPUs from Intel Socket 5/7 motherboards with stock BIOS

pankoza2-pl committed:
bring back the Slot 1 to Socket 8 adapter

pankoza2-pl committed:
Just block Pentium Pro on incompatibile machines, instead of removing the Slot 1-to-Socket 8 adapter altogether (part 2)

OBattler committed:
Bring machine flags on par with device flag and clean up device_is_valid() accordingly.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
Fixed (S)VGA memory access bitness stuff.

OBattler committed:
Removed the legacy #defines in device.h.

OBattler committed:
Corrected the ISA and Sidecar flags.

OBattler committed:
Assume device valid if it has no bus flags set.

OBattler committed:
Machine status: initialize it for the two cartridge bays as well.

OBattler committed:
Updated a header.

OBattler committed:
Merge pull request #5227 from pankozacorp/slotket8

Bring back the Slot 1 to Socket 8 adapter, but block it from use on incompatible machines.

OBattler committed:
Fixed the DEVICE_COM flags onwards, should fix serial mice.

OBattler committed:
Merge branch 'master' of https://github.com/86Box/86Box

OBattler committed:
Fixed DEVICE_BUS as well.
